### ISSUE TITLE
Ensure arity of result callbacks is consistent.

### ISSUE THREAD

**Issue by devinivy** [balderdashy/waterline#1046]
Per https://github.com/balderdashy/waterline/issues/1043#issuecomment-109657906, waterline can at times not be super friendly with the Async package.  Particularly to play well with `Async.waterfall`, it's important that results callbacks are always called with the same number of arguments, whether or not there were any actual results.  I think this primarily affects `findOne` and possibly `create`, since otherwise an empty result is typically `[]` (as in `find` and `update`).

Here's one example where this occurs for `findOne`: https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/finders/basic.js#L145

**Comment by bizarrochris** [comment]
upvote - now on line 155: https://github.com/balderdashy/waterline/blob/master/lib/waterline/query/finders/basic.js#L155

**Comment by sailsbot** [comment]
Thanks for posting, @devinivy.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
